src/libraries/Microsoft.PowerFx.Core/Parser/TexlParser.cs

```diff
@@ -317,7 +317,16 @@ private ParseUserDefinitionResult ParseUDFsAndNamedFormulas(string script, Parse
                         var result = ParseExpr(Precedence.None);
                         if (result is TypeLiteralNode typeLiteralNode)
                         {
-                            definedTypes.Add(new DefinedType(thisIdentifier.As<IdentToken>(), typeLiteralNode));
+                            if (TypeLiteralNode.IsValid(typeLiteralNode, out var err))

```

Comment:
>err [](http://example.com/codeflow?start=81&length=3)

Suggestion: since this can be multiple errors, call `errors` instead of `err` (which suggests singular)